core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2894,6 +2896,26 @@ public void testETagWithRegisterTable() {
     assertThat(respHeaders).containsEntry(HttpHeaders.ETAG, eTag);
   }
 
+  @Test
+  public void testLoadTableCredentials() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog = catalog(adapter);
+
+    catalog.createNamespace(TABLE.namespace());
+    catalog.createTable(TABLE, SCHEMA);
+
+    List<Credential> credentials = catalog.loadTableCredentials(TABLE);
```

Comment:
the reason I haven't added the reference impl back then when introducing the `/credentials` endpoint is because we can't really test it by going through the catalog, since the catalog would never call this endpoint in its normal flow. As I mentioned in https://github.com/apache/iceberg/pull/14749/files#r2591733910 the only places where this is called is in storage-specific code, which makes it difficult to add a test here